src/lib/useDragAndZoom.tsx

```diff
@@ -36,6 +37,22 @@ export function fitBoundsInLimit(bounds: Bounds, limit: Bounds | undefined): Bou
     return bounds;
 }
 
+export function fitBoundsInMinVisibleX (newBounds: Bounds, oldBounds: Bounds, minVisibleX: number = 1): Bounds {
+    if (newBounds[1] - newBounds[0] > minVisibleX) {
+        return newBounds;
+    }
+
+    const lenBounds = oldBounds[1] - oldBounds[0];
+
+    if (lenBounds <= minVisibleX) {
+        return oldBounds
+    }
+
+    const lenVisibleDiff = Math.abs(minVisibleX - lenBounds);
+
+    return [oldBounds[0] + lenVisibleDiff / 2, oldBounds[1] - lenVisibleDiff / 2]
```

Comment:
Seems like this can be simplified to:

```suggestion
    const center = (newBounds[0] + newBounds[1]) / 2;
    return [
        center - minVisibleX / 2,
        center + minVisibleX / 2
    ];
```